### PR TITLE
Instrument recent feature flows with Umami analytics

### PR DESCRIPTION
**Pull request by karanhudia**
## Description
Adds privacy-preserving Umami instrumentation for recent Borg UI feature surfaces:

- Cloud storage remotes and OAuth setup workflows.
- Managed agents, enrollment tokens, reinstall/copy/log/cancel actions.
- Backup plans, plan runs/history/logs, linked repository navigation, search, sort, grouping, and create/update/delete flows.

Also adds a dated engineering spec/plan and focused tests that mock `useAnalytics` to prove the new event boundaries emit stable non-sensitive metadata. The latest review-fix commit also makes Cloud Storage search analytics compute `result_count` from the submitted query and fixes markdown heading levels in the implementation plan.

## Related Issue
Linear: BOR-107 - https://linear.app/nullcodeai/issue/BOR-107/instrument-recent-borg-ui-features-with-umami-analytics

## Type of Change
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📖 Documentation update
- [ ] 🎨 UI/UX improvement
- [x] 🧪 Test addition/improvement
- [ ] ♻️ Refactoring

## Testing
- [x] I have tested this locally
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All existing tests pass

Validation run on the merged `origin/main` base:

- `cd frontend && npm run test -- --run src/pages/__tests__/CloudStorage.test.tsx --maxWorkers=1 --testTimeout=120000`
- `cd frontend && npm run test -- --run src/pages/__tests__/ManagedAgents.test.tsx --maxWorkers=1 --testTimeout=120000`
- `cd frontend && npm run test -- --run src/pages/backup-plans/__tests__/BackupPlansContent.test.tsx src/pages/__tests__/BackupPlans.test.tsx --maxWorkers=1 --testTimeout=60000`
- `cd frontend && npm run check:locales`
- `cd frontend && npm run typecheck -- --pretty false`
- `cd frontend && npm run lint`
- `cd frontend && npm run build`
- `./scripts/dev.sh` runtime route/API checks for `/cloud-storage`, `/managed-agents`, and `/backup-plans`

Additional review-fix validation on `ff86319`:

- `cd frontend && npm run test -- --run src/pages/__tests__/CloudStorage.test.tsx -t "tracks cloud storage search result counts" --maxWorkers=1 --testTimeout=60000`
- `git diff --check`
- `cd frontend && npm run check:locales`
- `cd frontend && npm run typecheck -- --pretty false`
- `cd frontend && npm run lint`
- `cd frontend && npm run build`

## Checklist
- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) guidelines
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] Comments were kept to existing patterns; no new complex code required explanatory comments
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

## Screenshots (if applicable)
Not applicable; this is analytics instrumentation with no UI rendering changes.

## Additional Context
Event payloads intentionally avoid raw remote names, backup plan names, repository names, filesystem paths, hostnames, tokens, OAuth secrets, and copied commands. Storybook stories/snapshots were not changed because no visible component state changed.

---

**Note:** By submitting this pull request, you agree that your contributions will be licensed under the GNU General Public License v3.0, the same license as this project.

## Summary by CodeRabbit

* **New Features**
  * Enhanced analytics tracking for Cloud Storage, Managed Agents, and Backup Plans to capture user interactions including create, edit, search, filter, and action workflows.

* **Documentation**
  * Added engineering plan and specification for analytics implementation across recent features.

* **Tests**
  * Added test coverage for analytics event tracking across Cloud Storage, Managed Agents, and Backup Plans workflows.

### CODE REVIEW

File: frontend/src/pages/CloudStorage.tsx [hunk 1]
```diff
@@ -584,10 +588,26 @@ export function CloudStorageContent({
...
     localStorage.setItem('cloud_storage_group', value)
+    trackSystem(EventAction.FILTER, {
+      section: CLOUD_STORAGE_ANALYTICS_SECTION,
+      operation: 'change_group',
+      sort_by: sortBy,
+      group_by: value,
+      query_length: searchQuery.trim().length,
+      result_count: totalAfterFilter,
+    })
   }
```
[the remark this answers is not in the document]
Comment (karanhudia) [reply frontend/src/pages/CloudStorage.tsx:621]: [codex] Fixed in ff86319. Cloud Storage analytics now computes result_count synchronously from the submitted search query, and the new focused regression test fails against the stale totalAfterFilter behavior.

File: docs/engineering/plans/2026-05-31-recent-feature-umami-events.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+**Goal:** Add Umami event coverage for recent cloud storage, managed agent, and backup plan features.
+
+**Architecture:** Instrument container pages where the workflow state and action handlers already live. Reuse `useAnalytics`, existing event categories, and stable non-sensitive payloads. Add focused tests by mocking `useAnalytics` in each page test file and asserting user actions call `track`/`trackSystem` with the expected metadata.
+
+**Tech Stack:** React, TypeScript, Vite, Vitest, React Testing Library, existing Borg UI analytics utilities.
+
+---
+
+### Task 1: Cloud Storage Analytics
```
[the remark this answers is not in the document]
Comment (karanhudia) [reply docs/engineering/plans/2026-05-31-recent-feature-umami-events.md]: [codex] Fixed in ff86319 by changing the task headings in the implementation plan from H3 to H2.

**Comment by karanhudia** [comment]
[codex] Informational GitHub Actions security and coverage comments reviewed; no code changes required. Continuing with the post-merge CI/checks gate for baba47b.

**Comment by karanhudia** [comment]
[codex] Updated coverage report reviewed: overall coverage is 70.5% and PR coverage is 100%. This is informational; no code changes required before merge.
